independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcContainer.java

```diff
@@ -233,4 +233,12 @@ public interface ArcContainer {
      * @see CurrentContext
      */
     CurrentContextFactory getCurrentContextFactory();
+
+    /**
+     * Indicates whether container runs in strict compatibility mode.
+     * Default value is false.
+     *
+     * @return true is strict mode is enabled, false otherwise.
+     */
+    boolean strictCompatibility();
```

Comment:
Do we want to make this a part of public API?

Reply:
The flag itself is part of the API since we use it to initialize the container. So I think that it's reasonable to add a read method as well.

Reply:
I originally didn't have it exposed but ended up doing that basically with the same reasoning Martin has here.
Not to mention it's nice not having to cast the result of `Arc.container()` to the impl class :)

Reply:
There's `ArcContainerImpl.instance()` to get the internal type, so that argument doesn't count IMHO :-) But if we need to expose that as a parameter to `Arc.initialize()`, then that makes sense. (For the record, I don't think we do, we could make it part of `ComponentsProvider` or something like that, so that we actually always propagate the build-time status to runtime instead of relying on the integrator to do it correctly. Of course that's a bit more complex.)

Reply:
> (For the record, I don't think we do, we could make it part of ComponentsProvider or something like that, so that we actually always propagate the build-time status to runtime instead of relying on the integrator to do it correctly. Of course that's a bit more complex.)

So this is actually what I started looking into but after chat with Martin, I landed on current solution.